homeassistant/components/adax/sensor.py

```diff
@@ -0,0 +1,83 @@
+"""Support for Adax energy sensors."""
+
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfEnergy
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Adax energy sensors from a config entry."""
+    adax_data_handler = hass.data[entry.entry_id]
+    sensors = []
+
+    for room in adax_data_handler.get_rooms():
+        sensors.append(AdaxEnergySensor(adax_data_handler, room))
+
+    async_add_entities(sensors, True)
+
+
+class AdaxEnergySensor(SensorEntity):
+    """Representation of an Adax Energy Sensor."""
+
+    def __init__(self, adax_data_handler, room):
+        """Initialize the sensor."""
+        self._adax_data_handler = adax_data_handler
+        self._heater_data = room
+        self._state = None
+
+    @property
+    def name(self):
```

Comment:
Use self._attr_name . Same with most of the other properties

https://developers.home-assistant.io/docs/core/entity?_highlight=_attr_#entity-class-or-instance-attributes

Reply:
I changed to use _attr_ for name, unique_id, device_class and state_class